src/setup-java.ts

```diff
@@ -19,12 +20,52 @@ async function run() {
     const checkLatest = getBooleanInput(constants.INPUT_CHECK_LATEST, false);
     let toolchainIds = core.getMultilineInput(constants.INPUT_MVN_TOOLCHAIN_ID);
 
+    core.startGroup('Installed distributions');
+
     if (versions.length !== toolchainIds.length) {
       toolchainIds = [];
     }
 
-    core.startGroup('Installed distributions');
+    if (!versions.length) {
+      core.debug('JAVA_VERSION input is empty, looking for .java-version file');
+      const versionFileName = '.java-version';
+      const contents = fs
+        .readFileSync(versionFileName)
+        .toString()
+        .trim();
+      const semverRegExp = /(?<version>(?<=(^|\s|\-))(\d+\S*))(\s|$)/;
+      let version = contents.match(semverRegExp)?.groups?.version
+        ? (contents.match(semverRegExp)?.groups?.version as string)
+        : '';
+      let installed = false;
+      while (!installed && version != '') {
+        try {
+          core.debug(`Trying to install version ${version}`);
+          await installVersion(version);
+          installed = true;
+        } catch (error) {
+          core.debug(`${error.toString()}`);
+          version = getHigherVersion(version);
```

Comment:
I don't think we need the `getHigherVersion` function and this loop. If I specify the full version (for example like this `15.0.0-ea.15`), then I expect that exactly this version will be installed, and I prefer to get an error message if it doesn't happen. The current approach is not obvious and I (as a user) may not notice that higher version is installed.

Reply:
After discussion with colleagues, we decided that this behavior would be more convenient, as this way we can guarantee a greater likelihood of installation success. Of course, this behavior will be described in the documentation, so it will not come as a surprise to users.